Source/WebCore/rendering/RenderBoxModelObject.cpp

```diff
@@ -1218,9 +1218,13 @@ BackgroundImageGeometry RenderBoxModelObject::calculateBackgroundImageGeometry(c
     LayoutSize positioningAreaSize;
     // Determine the background positioning area and set destination rect to the background painting area.
     // Destination rect will be adjusted later if the background is non-repeating.
-    // FIXME: transforms spec says that fixed backgrounds behave like scroll inside transforms. https://bugs.webkit.org/show_bug.cgi?id=15679
     LayoutRect destinationRect(borderBoxRect);
-    bool fixedAttachment = fillLayer.attachment() == FillAttachment::FixedBackground;
```

Comment:
This code has moved into `BackgroundPainter::calculateBackgroundImageGeometry`, would you be able to rebase against latest trunk?

I'd also expect `LayoutTests/imported/w3c/web-platform-tests/css/css-transforms/transform-fixed-bg-002/4/5/6/7.html` to pass with this change, do you need to update the `TestExpectations` file?

Other than that, looks good to me!

Reply:
rebased